setup/Setup/SetupOperation.cs

```diff
@@ -129,16 +129,25 @@ protected class WorkItem
 			}
 		}
 
-		public static bool DeleteEmptyDirs(string dir) {
-			bool allEmpty = true;
-			foreach (var subDir in Directory.EnumerateDirectories(dir))
-				allEmpty &= DeleteEmptyDirs(subDir);
-			
-			if (!allEmpty || Directory.EnumerateFiles(dir).Any())
-				return false;
-
-			Directory.Delete(dir);
-			return true;
+		public static bool DeleteEmptyDirs(string directory) {
+			if (!Directory.Exists(directory))
+				return true;
+
+			static bool Recursion(string dir) {
+				bool allEmpty = true;
+
+				foreach(string subDir in Directory.EnumerateDirectories(dir))
+					allEmpty &= Recursion(subDir);
+
+				if(!allEmpty || Directory.EnumerateFiles(dir).Any())
+					return false;
+
+				Directory.Delete(dir);
+
+				return true;
+			}
+
+			return Recursion(directory);
```

Comment:
Can't this be resolved a bit simpler?
```suggestion
		public static bool DeleteEmptyDirs(string directory) {
			if (!Directory.Exists(directory)) return true;
			
			bool allEmpty = true;
			
			foreach(string subDir in Directory.EnumerateDirectories(directory))
				allEmpty &= DeleteEmptyDirs(subDir);
			
			if (!allEmpty || Directory.EnumerateFiles(dir).Any()) return false;
			
			Directory.Delete(directory);
			return true;
		}
```

Reply:
You're asking to revert my second commit, which just got rid of useless Directory.Exists checks past the first invocation. I'll just rewrite this method, sec.